lib/melodica_inventory_web/plugs/authenticate.ex

```diff
@@ -1,12 +1,14 @@
 defmodule MelodicaInventoryWeb.Plugs.Authenticate do
+  @moduledoc false
+
   import Plug.Conn
   import Phoenix.Controller
   alias MelodicaInventoryWeb.Router.Helpers, as: RouterHelpers
 
   def init(default), do: default
 
   def call(%Plug.Conn{assigns: %{current_user: current_user}} = conn, :admin) do
-    case !!current_user and current_user.admin do
+    case present?(current_user) and current_user.admin do
```

Comment:
This is because `and` expects a boolean value, while `&&` works with truthy and falsey values.